app/org/sagebionetworks/bridge/dynamodb/DynamoSchedulePlanDao.java

```diff
@@ -88,39 +95,103 @@ public SchedulePlan getSchedulePlan(StudyIdentifier studyIdentifier, String guid
     }
 
     @Override
     public SchedulePlan updateSchedulePlan(StudyIdentifier studyIdentifier, SchedulePlan plan) {
-        checkNotNull(studyIdentifier, "StudyIdentifier is null");
-        checkNotNull(plan, "Schedule plan is null");
+        checkNotNull(studyIdentifier);
+        checkNotNull(plan);
         
         plan.setStudyKey(studyIdentifier.getIdentifier());
         plan.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
+        
+        forEachCriteria(plan, scheduleCriteria -> persistCriteria(scheduleCriteria));
         mapper.save(plan);
         return plan;
     }
 
     @Override
     public void deleteSchedulePlan(StudyIdentifier studyIdentifier, String guid) {
-        checkNotNull(studyIdentifier, "Study identifier is null");
-        checkArgument(StringUtils.isNotBlank(guid), "Plan GUID is blank or null");
+        checkNotNull(studyIdentifier);
+        checkArgument(isNotBlank(guid));
         
         SchedulePlan plan = getSchedulePlan(studyIdentifier, guid);
+        
+        forEachCriteria(plan, scheduleCriteria -> deleteCriteria(scheduleCriteria));
         mapper.delete(plan);
     }
+    
+    private void forEachCriteria(SchedulePlan plan, Function<ScheduleCriteria,Criteria> consumer) {
+        if (plan.getStrategy() instanceof CriteriaScheduleStrategy) {
+            CriteriaScheduleStrategy strategy = (CriteriaScheduleStrategy)plan.getStrategy();
+            for (int i=0; i < strategy.getScheduleCriteria().size(); i++) {
+                ScheduleCriteria scheduleCriteria = strategy.getScheduleCriteria().get(i);
+                
+                // Add the key in order to load
+                scheduleCriteria.getCriteria().setKey(getKey(plan, i));
+
+                Criteria criteria = consumer.apply(scheduleCriteria);
+                
+                // Update the criteria object (except for delete). This may add a new criteria object, 
+                // which will return the plan to the caller with the criteria stubbed out in the JSON.
+                if (criteria != null) {
+                    scheduleCriteria = new ScheduleCriteria(scheduleCriteria.getSchedule(), criteria);
+                    strategy.getScheduleCriteria().set(i, scheduleCriteria);
+                }
+            }
+        }        
+    }
+    
+    private String getKey(SchedulePlan plan, int index) {
+        return "scheduleCriteria:" + plan.getGuid() + ":" + index;
+    }
+    
+    /**
+     * Save the criteria object if it exists. If not, return an empty criteria object which will return the 
+     * criteria stubbed out in the JSON representation of the schedule plan.
+     */
+    private Criteria persistCriteria(ScheduleCriteria scheduleCriteria) {
+        Criteria criteria = scheduleCriteria.getCriteria();
+        if (criteria != null) {
+            criteriaDao.createOrUpdateCriteria(criteria);
+        }
+        return (criteria != null) ? criteria : Criteria.create();
+    }
```

Comment:
You could submit a schedule plan where the criteria property is null, in order to indicate there are no criteria. That's okay, we do set an empty criteria object though to reduce the possibility of errors.